src/pveclib/vec_f128_ppc.h

```diff
@@ -951,17 +1314,352 @@ test_cmpltf128_v3d (vui128_t vfa128, vui128_t vfb128)
+	    }
+	}
+      else
+	{ // less than 1.0 or negative
+	  result = (vui128_t) q_zero;
+	}
+    }
+  else
+    { // isinf or isnan.
+      vb128_t is_inf;
+      // Positive Inf returns all ones
+      // else NaN or -Infinity returns zero
+      is_inf = vec_cmpequq (q_sig, (vui128_t) q_zero);
+      // result = ~NaN | (pos & Inf) -> Inf & (pos & Inf) -> pos & Inf
+      result = (vui128_t) vec_andc ((vui32_t) is_inf, (vui32_t) b_sign);
+    }
+ * \endcode
+ * As is the usual for floating-point conversions, we extract the sign,
+ * significand, and exponent then test for class and range.
+ * We compare the extracted exponent directly using vector doubleword
+ * compares. These are faster (on POWER8) than quadword compares but
+ * require doubleword splatting the QP exponent and compare constants
+ * for correct results. This only requires one additional instruction
+ * (xxpermdi) as the vector constants will be loaded as quadwords
+ * either way.
+ *
+ * The outter test is for NaN/Infinity. These should be rare so we use
+ * __builtin_expect().  The implementation returns special values
+ * to match the instruction definition.
+ *
+ * Once we know the value is finite, we check for greater than or equal
+ * to +1.0. Negative or fractional values return quadword zero.
+ * Then we check for less than 2<sup>128</sup>. If not we return
+ * all ones (2<sup>128</sup> -1).
+ *
+ * If the input is in the valid range for unsigned quadword we
+ * left-justify the significand then shift the quadword right by
+ * (127 - <sub>unbiased</sub>exp). The right shift truncates
+ * (round toward zero) any fractional bits. See vec_xscvqpuqz().
+ *
+ * The signed operation follows similar logic with appropriate
+ * adjustments for negative values and reduced magnitude range.
+ * The doubleword versions of the convert operation follows the
+ * same outline with different range constants.
+ * See vec_xscvqpsqz(), vec_xscvqpudz() and vec_xscvqpsdz().
+ *
+ * \subsubsection f128_softfloat_0_0_2_3 Convert Quadword integer to Quad-Precision
  * TBD
  *
- * \note See
- * <a href="https://www.exploringbinary.com/gcc-avoids-double-rounding-errors-with-round-to-odd/">
-*  GCC Avoids Double Rounding Errors With Round-To-Odd</a>
+ * Conversions from doubleword integer to quad-precision float can be
+ * represented exactly and do not require any rounding.
+ * But conversions from quadword integer to quad-precision float may
+ * overflow the 113-bit significand which does require rounding.
+ *
+ * For POWER9 we have the
+ * <B>VSX Scalar Convert Signed/Unsigned Doubleword to Quad-Precision format
+ * <I>(xscvsdqp/xscvudqp)</I></B> instructions.
+ * For POWER10 we have the
+ * <B>VVSX Scalar Convert with round Signed/Unsigned Quadword to
+ * Quad-Precision format
+ * <I>(xscvsqqp/xscvuqqp)</I></B> instructions.
+ * One of four rounding modes is selected from the 2-bit
+ * <B>FPSCR.<sub>RN</sub></B> field.
+ * The default rounding mode is <I>Round to Nearest, ties to even</I>
```

Comment:
In the Power ISA, this is called "Round to Nearest Even"

Reply:
Ok the PowerISA is mixed on this. PowerISA 3.1 is 19 counts of "Round to Nearest Even" and and 8 for "Round to Nearest, Ties to Even". IEEE-754-2008 uses roundTiesToEven.

But I am willing to use "Round to Nearest Even" because it is shorter.